pkg/controllers/edgeconnect/controller.go

```diff
@@ -617,27 +625,18 @@ func (controller *Controller) updateEdgeConnect(ctx context.Context, edgeConnect
 }
 
 func (controller *Controller) createOrUpdateEdgeConnectDeployment(ctx context.Context, edgeConnect *edgeconnectv1alpha1.EdgeConnect) error {
-	clientSecretName := edgeConnectClientSecretName(edgeConnect.Name)
+	clientSecretName := edgeConnect.ClientSecretName()
 
 	_log := log.WithValues("namespace", edgeConnect.Namespace, "name", edgeConnect.Name, "clientSecretName", clientSecretName)
 
-	secretQuery := k8ssecret.NewQuery(ctx, controller.client, controller.apiReader, _log)
-
-	secret, err := secretQuery.Get(types.NamespacedName{Name: clientSecretName, Namespace: edgeConnect.Namespace})
+	secretHash, err := controller.createOrUpdateEdgeConnectConfigSecret(ctx, edgeConnect)
 	if err != nil {
-		_log.Debug("EdgeConnect client secret not found")
-
 		return err
 	}
 
-	resource, err := k8ssecret.ExtractToken(&secret, consts.KeyEdgeConnectOauthResource)
-	if err != nil {
-		_log.Debug("Unable to extract token from EdgeConnect client secret")
-
-		return err
-	}
+	desiredDeployment := deployment.New(edgeConnect)
 
-	desiredDeployment := deployment.NewProvisioner(edgeConnect, clientSecretName, resource)
+	desiredDeployment.Spec.Template.Annotations = map[string]string{consts.EdgeConnectAnnotationSecretHash: secretHash}
```

Comment:
I couldn't find where you calculate the hash for the desired deployment, so maybe I am going to say something dumb. 😅
Shouldn't we have this annotation part of the desired deployment when the hash is generated for it? 🤔

We want to have a hash on the deployment so we don't redeploy the resource in case something (a webhook) dynamically changed it.
We want to have a hash for the secret so if the secret changes, so we redeploy the deployment if the secret changes.
- If we have the secret hash part of the desired deployment at the time of the hash calculation of the deployment, then you only have to check the hash of the deployment to know if a redeployment is needed.

Reply:
we do it here https://github.com/dynatrace/dynatrace-operator/blob/b6f2ac5ef9dc4214034b12415d9475d36e94e9ef/pkg/controllers/edgeconnect/controller.go#L332-L340, as I already replied to @luhi-DT  in slack, it does't work if I add annotation of secret data to deployment.

I see new secret hash updated, but it doesn't restart pod of deployment.
```
[50](https://dynatrace.slack.com/archives/D05FG9U937D/p1712569856583619)
│                         internal.operator.dynatrace.com/template-hash: 2620308800                                                            │
│                         my-secret-hash: 855148224
```
apply few changes to secret in between:
```
│                         [internal.operator.dynatrace.com/template-hash](http://internal.operator.dynatrace.com/template-hash): 1846043313                                                            │
│                         my-secret-hash: 2079240476
```